src/azure-cli-core/azure/cli/core/commands/template_create.py

```diff
@@ -8,11 +8,11 @@
 
 import azure.cli.core.azlogging as azlogging
 from azure.cli.core.util import CLIError
-from azure.cli.core.commands.arm import (
+from azure.cli.core.commands.arm import resource_exists
+from msrestazure.tools import (
     is_valid_resource_id,
     parse_resource_id,
-    resource_id,
-    resource_exists)
+    resource_id)
```

Comment:
Same here

Reply:
There's definitely many cases in the code base where something is imported from msrestazure in the top level(namely, CloudError):
https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py#L14

Should I still go about only importing the parsing tools in the methods they are used in? I just want a confirmation before I go about doing this, as there are many locations that I will have to make the change.

Reply:
All the `msrestazure` imports in `azure-cli-core` should be at the method level rather than at the very top.

This search below shows all the places we import msrestazure in azure-cli-core and they're all inside the methods.
The imports in custom.py don't matter as much 
https://github.com/search?utf8=%E2%9C%93&q=from+msrestazure+repo%3AAzure%2Fazure-cli+path%3Asrc%2Fazure-cli-core&type=Code&ref=advsearch&l=&l=

And this is why we should move the imports:
```
$ time python -c 'import msrestazure.tools'

real    0m0.271s
user    0m0.231s
sys     0m0.037s
```

Reply:
k, will do